File: examples/notify.R
library(shiny)

ui <- fluidPage(
  tags$h2("notify examples"),
  tags$p(
    "More examples available on the official website:",
    tags$a("https://notiflix.github.io/notify")
  ),
  actionButton("success", "Success"),
  actionButton("failure", "Failure"),
  actionButton("info", "Info"),
  actionButton("warning", "Warning")
)

server <- function(input, output, session) {

  observeEvent(input$success, {
    notify_success("Well done!")
  })

  observeEvent(input$failure, {
    notify_failure("Oups...")
  })

  observeEvent(input$info, {
    notify_info("For your information")
  })

  observeEvent(input$warning, {
    notify_warning("Be careful!")
  })

}

if (interactive())
  shinyApp(ui, server)

File: srcjs/exts/notify.js
import "shiny";
import { Notify } from "notiflix/build/notiflix-notify-aio";

Shiny.addCustomMessageHandler("shinybusy-notify-success", function(data) {
  Notify.success(data.text, data.config);
});
Shiny.addCustomMessageHandler("shinybusy-notify-failure", function(data) {
  Notify.failure(data.text, data.config);
});
Shiny.addCustomMessageHandler("shinybusy-notify-warning", function(data) {
  Notify.warning(data.text, data.config);
});
Shiny.addCustomMessageHandler("shinybusy-notify-info", function(data) {
  Notify.info(data.text, data.config);
});

